nodejs/week3/homework/warmup/index.js

```diff
@@ -0,0 +1,183 @@
+const app = express();
+
+// Parse URL-encoded bodies (as sent by HTML forms)
+app.use(express.urlencoded({ extended: true }));
+// Parse JSON bodies (as sent by API clients)
+app.use(express.json());
+
+// get methods:
+app.get('/calculator', (req, res) => {
+    const firstParam = req.query.firstParam;
+    const secondParam = req.query.secondParam;
+    const firstParamErrorMsg = ' first param is not a number';
+    const secondParamErrorMsg = ' the second param is not a number';
+    let answer;
+
+    if (isNaN(firstParam)) {
+        return res.send(firstParamErrorMsg);
+    }
+
+    if ('addition' in req.query) {
+        if (typeof secondParam === 'object') {
+            answer =
+                secondParam.reduce((one, another) => Number(one) + Number(another)) +
+                Number(firstParam);
+
+            if (isNaN(answer)) {
+                return res.send(secondParamErrorMsg);
+            }
+            return res.json(answer);
+        }
+        if (isNaN(secondParam)) {
+            return res.send(secondParamErrorMsg);
+        }
+        answer = Number(firstParam) + Number(secondParam);
+
+        res.json(answer);
+    }
+
+    if ('substraction' in req.query) {
+        if (typeof secondParam === 'object') {
+            answer = secondParam.reduce(
+                (one, another) => Number(one) + Number(another),
+            );
+
+            if (isNaN(answer)) {
+                return res.send(secondParamErrorMsg);
+            }
+            return res.json(Number(firstParam) - answer);
+        }
+        if (isNaN(secondParam)) {
+            return res.send(secondParamErrorMsg);
+        }
+        answer = Number(firstParam) - Number(secondParam);
+
+        res.json(answer);
+    }
+
+    if ('multiplication' in req.query) {
+        if (typeof secondParam === 'object') {
+            answer =
+                secondParam.reduce((one, another) => Number(one) * Number(another)) *
+                Number(firstParam);
+
+            if (isNaN(answer)) {
+                return res.send(secondParamErrorMsg);
+            }
+            return res.json(answer);
+        }
+        if (isNaN(secondParam)) {
+            return res.send(secondParamErrorMsg);
+        }
+        answer = Number(firstParam) * Number(secondParam);
+
+        res.json(answer);
+    }
+    if ('division' in req.query) {
+        if (typeof secondParam === 'object') {
+            answer = secondParam.reduce(
+                (one, another) => Number(one) * Number(another),
+            );
+            if (isNaN(answer)) {
+                return res.send(secondParamErrorMsg);
+            }
+            return res.json(Number(firstParam) / answer);
+        }
+        if (isNaN(secondParam)) {
+            return res.send(secondParamErrorMsg);
+        }
+        answer = Number(firstParam) / Number(secondParam);
+
+        res.json(answer);
+    }
+});
+
+//  post method
+app.post('/calculator', (req, res) => {
```

Comment:
Nice job adding the POST request and getting the parameters from the request's body. 
However, I suggest you split them to separate routes as well:
```
app.post('/calculator/addition', (req, res)
```